### PR TITLE
Setters for the tobin tax

### PR DESCRIPTION
**Pull request by rcroessmann**
## Description

Creates setters for the tobin tax and the ratio at which the tobin tax kicks in.

## Tested

No

### CODE REVIEW

File: packages/protocol/contracts/stability/Reserve.sol [hunk 1]
```diff
@@ -109,6 +112,15 @@ contract Reserve is IReserve, Ownable, Initializable, UsingRegistry, ReentrancyG
     emit TobinTaxSet(value);
   }
 
+  /**
+   * @notice Sets the tobin tax.
+   * @param value The tobin tax.
+   */
+  function setTobinTax(uint256 value) public onlyOwner {
```
Comment (rcroessmann) [inline packages/protocol/contracts/stability/Reserve.sol]: Copy pasted this in again by accident, removed this duplicate block in next commit.

File: packages/protocol/contracts/stability/Reserve.sol [hunk 2]
```diff
@@ -438,15 +469,14 @@ contract Reserve is IReserve, Ownable, Initializable, UsingRegistry, ReentrancyG
...
   /**
    * @notice Computes a tobin tax based on the reserve ratio.
-   * @return The numerator of the tobin tax amount, where the denominator is 1000.
+   * @return The tobin tax expresesed as a fixidity fraction.
    */
   function computeTobinTax() private view returns (FixidityLib.Fraction memory) {
-    // The protocol calls for a 0.5% transfer tax on Celo Gold when the reserve ratio <= 2.
     FixidityLib.Fraction memory ratio = FixidityLib.wrap(getReserveRatio());
-    if (ratio.gte(FixidityLib.newFixed(2))) {
+    if (ratio.gte(FixidityLib.wrap(tobinTaxReserveRatio))) {
```
Comment (yorhodes) [inline packages/protocol/contracts/stability/Reserve.sol:476]: any reason why `tobinTaxReserveRatio` isn't also a Fixidity frac?

File: packages/protocol/contracts/stability/Reserve.sol [hunk 3]
```diff
@@ -438,15 +469,14 @@ contract Reserve is IReserve, Ownable, Initializable, UsingRegistry, ReentrancyG
...
       return FixidityLib.wrap(0);
     } else {
-      return FixidityLib.wrap(TOBIN_TAX_NUMERATOR);
+      return FixidityLib.wrap(tobinTax);
```
Comment (yorhodes) [inline packages/protocol/contracts/stability/Reserve.sol:479]: ditto with `tobinTax`? we would save gas by computing this in the setters

**Review by yorhodes — APPROVED** [review]

File: packages/protocol/contracts/stability/Reserve.sol (hunk 3 above)
Comment (yorhodes) [reply packages/protocol/contracts/stability/Reserve.sol:479]: in fact we do compute it in the setter but don't store the value

Comment (yorhodes) [reply packages/protocol/contracts/stability/Reserve.sol:479]: (mixed up `wrap` and `newFixed` but we would still save some albeit minimal amt of gas

File: packages/protocol/contracts/stability/Reserve.sol (hunk 2 above)
Comment (rcroessmann) [reply packages/protocol/contracts/stability/Reserve.sol:476]: Given the timing, I tried to stay as close as possible to the previous implementation.

File: packages/protocol/contracts/stability/Reserve.sol (hunk 3 above)
Comment (rcroessmann) [reply packages/protocol/contracts/stability/Reserve.sol:479]: See comment above.

**Pull request merged**